src/RF24Log.h

```diff
@@ -0,0 +1,274 @@
+/** This level is for warning outputs and will output messages of @ref WARN & @ref ERROR levels */
+#define WARN    30
+/** This level is for error outputs and will output messages of only @ref ERROR levels */
+#define ERROR   40
+
+/**
+ * @}
+ * A templated class for implementing a single logging object.
+ * Copied @ref logging objects (using RF24Logger(RF24Logger*)) will re-use
+ * the original handler and level.
+ * @see the pre-instantiated @ref logging variable
+ */
+template <typename StreamType>
+class RF24Logger
+{
+private:
+
+    uint8_t level; /** the logging level used to filter logging messages */
+    StreamType* handler; /** the output stream */
+    char _name[40]; /** the logger instance's name (max of 39 characters)*/
+
+public:
+
+    /** Empty constructor. level defaults to @ref NOT_SET, and instance has no handler. */
+    RF24Logger() : level(NOT_SET), handler(nullptr)
+    {
+        _name[0] = 0;
+    }
+
+    /**
+     * Instance constructor. The , and the instance's
+     * handler is initialized.
+     * @param stream The handler to which all logging output will be directed.
+     * @param name The origin's name of the logger's messages.
+     * @param lvl The instance's logging level defaults to @ref NOT_SET if not specified.
+     */
+    RF24Logger(StreamType* stream, const char* name, uin8_t lvl = NOT_SET) : level(lvl), handler(stream)
+    {
+        strcpy(_name, name);
+    }
+
+    /**
+     * Copy constructor. Instance's log level and handler are set to @p obj instance's
+     * corresponding values.
+     * @param obj An instantiated RF24Logger object from which values are copied from.
+     */
+    RF24Logger(RF24Logger* obj) : level(obj->level), handler(obj->handler)
+    {
+        _name[0] = 0;
+    }
+
+    /**
+     * Set the handler to which all logging messages are directed.
+     * @param stream The output stream to be used as the handler.
+     */
+    void setHandler(StreamType* stream)
+    {
+        handler = stream;
+    }
+
+    /**
+     * Set the logging level that's to filter logging messages passed to log()
+     * @param lvl The logging level must be in range [0, 60).
+     */
+    void setLevel(uint8_t lvl)
+    {
+        level = rf24_min(lvl, ERROR + 9);
+    }
+
+    /**
+     * @brief Set a default @p name for the instance
+     *
+     * @param name This will be used in calls to log(uint8_t lvl, Ts msg...)
+     */
+    void setName(const char* name)
+    {
+        strcpy(_name, name);
+    }
+
+    /**
+     * Returns the logging handler that was configured with RF24Logger(RF24Logger*),
+     * RF24Logger(StreamType*, const char*) or setHandler()
+     *
+     * Example usage:
+     * @code
+     * logging.getLogger() << "a string of text " << 0 << ':' << 1.0 << endl;
+     * @endcode
+     * would ouput "a string of text 0:1.0" with a trailing line feed.
+     * @returns The instantiated output stream object passed to setHandler()
+     */
+    StreamType &getLogger()
```

Comment:
this is a relic of experimenting. I agree because the `PrintOut` obj could be used directly for the same purpose. BTW there's also a `PrintOut::get()` function that should go also.

Reply:
Ok.